decider-Estimate/src/main/java/ca/on/oicr/pde/deciders/EstimateDecider.java

```diff
@@ -0,0 +1,391 @@
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import net.sourceforge.seqware.common.hibernate.FindAllTheFiles.Header;
+import net.sourceforge.seqware.common.module.FileMetadata;
+import net.sourceforge.seqware.common.module.ReturnValue;
+import net.sourceforge.seqware.common.util.Log;
+
+/**
+ *
+ * @author prath@oicr.on.ca
+ */
+public class EstimateDecider extends OicrDecider {
+
+    private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S");
+    private Map<String, BeSmall> fileSwaToSmall;
+
+    private String templateType = "WT";
+    private String queue = "";
+    private String studyTitle;
+
+    private final static String TXT_METATYPE = "text/plain";
+//    private String tumorType;
+//    private List<String> results;
+
+    public EstimateDecider() {
+        super();
+        fileSwaToSmall = new HashMap<String, BeSmall>();
+        parser.acceptsAll(Arrays.asList("ini-file"), "Optional: the location of the INI file.").withRequiredArg();
+        parser.accepts("template-type", "Required. Set the template type to limit the workflow run "
+                + "so that it runs on data only of this template type").withRequiredArg();
+        parser.accepts("queue", "Optional: Set the queue (Default: not set)").withRequiredArg();
+        parser.accepts("tumor-type", "Optional: Set tumor tissue type to something other than primary tumor (P), i.e. X . Default: Not set (All)").withRequiredArg();
+        parser.accepts("study-name", "Required. Specify study name, e.g. TGL07, OCT").withRequiredArg();
+    }
+
+    @Override
+    public ReturnValue init() {
+        Log.debug("INIT");
+        this.setMetaType(Arrays.asList(TXT_METATYPE));
+        this.setHeadersToGroupBy(Arrays.asList(Header.FILE_SWA));
+
+        ReturnValue rv = super.init();
+        rv.setExitStatus(ReturnValue.SUCCESS);
+
+        //Group by sample if no other grouping selected
+        if (this.options.has("group-by")) {
+            Log.error("group-by parameter passed, but this decider does not allow overriding the default grouping (by Donor + Library Type)");
+        }
+
+        if (this.options.has("queue")) {
+            this.queue = options.valueOf("queue").toString();
+        }
+
+        if (this.options.has("template-type")) {
+            if (!options.hasArgument("template-type")) {
+                Log.error("--template-type requires an argument, WT");
+                rv.setExitStatus(ReturnValue.INVALIDARGUMENT);
+                return rv;
+            } else {
+                this.templateType = options.valueOf("template-type").toString();
+                if (!this.templateType.equals("WT")) {
+                    Log.stderr("NOTE THAT ONLY WT template-type SUPPORTED, WE CANNOT GUARANTEE MEANINGFUL RESULTS WITH OTHER TEMPLATE TYPES");
+                    rv.setExitStatus(ReturnValue.INVALIDARGUMENT);
```

Comment:
This is an assumption that will break production someday. I suggest giving a warning but NOT exiting.